scripts/loki_transform.py

```diff
@@ -236,6 +233,10 @@ def convert(
         scheduler.process( HoistTemporaryArraysAnalysis(dim_vars=(vertical.size,)) )
         scheduler.process( SCCHoistTemporaryArraysTransformation(block_dim=block_dim) )
 
+    scheduler.process( SCCAnnotateTransformation(
+            horizontal=horizontal, vertical=vertical, directive=directive, block_dim=block_dim
+    ))
+
     if mode in ['cuf-parametrise', 'cuf-hoist', 'cuf-dynamic']:
```

Comment:
Notice that this change moves the location of `SCCAnnotate` "one step" further in the SCC pipeline. The reasoning for this is that `SCCHoistTemporaryArraysTransformation` does the SCC hoisting, introducing new arguments to subroutines, that have to be marked as "present", This change ensures that `SCCAnnotate` adds these pragmas for the new arguments as well.

I am not sure whether this should actually be moved after the next if statement even.